content/articles/The-concept-of-cookies-in-servlets/index.md

```diff
@@ -0,0 +1,222 @@
+We'll develop two servlet classes in this example, one to create cookies and the other to retrieve them. Here we will create two cookies to hold the user name and the allocated user subject, respectively.
+
+In this scenario, the JSP page gives us a form to fill out with our information, such as user name and subject.
+
+#### Steps to create this Cookies project:
+
+#### Step 1: Creating the Example to demonstrate cookies in servlet project.
+Open Eclipse or any other Java IDE, create a new Dynamic web project by selecting a new Dynamic web project. Provide your favorite project name. In my case, I will call it `Example to demonstrate cookies in servlet`.
+#### Step 2: Creating index.JSP page 
+By right-clicking on `Example to demonstrate cookies in servlet` and selecting New JSP, you can code a JSP page as shown. Provide your JSP class name. I will call it `index.jsp`.
+
+```JSP
+
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta charset="ISO-8859-1">
+    <title>Insert title here</title>
+</head>
+
+<body>
+    <form action="ProcessOneServlet">
+        Enter User name <input type="text" name="userName" required><br>
+
+        Enter subject <input type="text" name="subject" required><br>
+        <input type="submit" value="Click me">
+    </form>
+</body>
+
+</html>
+```
+
+#### Step 3: Creating a ProcessOneServlet Servlet page
+By right-clicking on the`Example to demonstrate cookies in servlet` project selecting New Servlet, you can code a Servlet page as shown below. Provide your Servlet class name. I will call it `ProcessOneServlet`.
+
+```java
+
+
+import java.io.IOException;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/ProcessOneServlet")
+public class ProcessOneServlet extends HttpServlet 
+{
+	private static final long serialVersionUID = 1L;
+    
+	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException 
+  {
+		response.setContentType("text/html");
+		String uName=request.getParameter("userName");
+		String subject=request.getParameter("subject");
+		Cookie object=new Cookie("cookieuName",uName);
+		Cookie object2=new Cookie("cookieSubject",subject);
+		object.setMaxAge(365*24*60*60);
+		response.addCookie(object);
+		object2.setMaxAge(365*24*60*60);
+		response.addCookie(object2);
+		RequestDispatcher req=request.getRequestDispatcher("ReadingValuesFromCookies");
+		req.forward(request, response);
+	}
+
+}
+```
+#### Step 3: Creating a ReadingValuesFromCookies Servlet page
+By right-clicking on the`Example to demonstrate cookies in servlet` project selecting New Servlet, you can code a Servlet page as shown below. Provide your Servlet class name. I will call it `ReadingValuesFromCookies`.

```

Comment:
Change to "You can code a Servlet page as shown below by right-clicking on the`Example to demonstrate cookies in servlet` project selecting New Servlet. Provide your Servlet class name. I will call it `ReadingValuesFromCookies`."